protocol-units/execution/opt-executor/src/executor/execution.rs

```diff
@@ -41,51 +41,50 @@ impl Executor {
 			}
 		};
 
+		let block_executor = self.block_executor.clone();
+
 		let block_id = block.block_id.clone();
-		let parent_block_id = {
-			let block_executor = self.block_executor.read().await;
-			block_executor.committed_block_id()
-		};
+		let parent_block_id = block_executor.committed_block_id();
 
 		let mempool_transactions = block.transactions.clone().into_txns();
 
-		let state_compute = {
-			let block_executor = self.block_executor.write().await;
-			block_executor.execute_block(
+		let block_executor_clone = block_executor.clone();
+		let state_compute = tokio::task::spawn_blocking(move || {
+			block_executor_clone.execute_block(
 				block,
 				parent_block_id,
 				BlockExecutorConfigFromOnchain::new_no_block_limit(),
-			)?
-		};
+			)
+		}).await??;
+
 		debug!("Block execution compute the following state: {:?}", state_compute);
 
 		let version = state_compute.version();
 		debug!("Block execution computed the following version: {:?}", version);
 		let (epoch, round) = (block_metadata.epoch(), block_metadata.round());
-
-		{
-			let ledger_info_with_sigs = self.ledger_info_with_sigs(
-				epoch,
-				round,
-				block_id.clone(),
-				block_metadata.timestamp_usecs(),
-				state_compute.root_hash(),
-				version,
-			);
-			let block_executor = self.block_executor.write().await;
-			block_executor.commit_blocks(vec![block_id], ledger_info_with_sigs)?;
-		}
-
-		// commit mempool transactions
-		{
-			let mut core_pool = self.core_mempool.write().await;
-			for transaction in mempool_transactions.iter() {
+	
+		let ledger_info_with_sigs = self.ledger_info_with_sigs(
+			epoch,
+			round,
+			block_id.clone(),
+			block_metadata.timestamp_usecs(),
+			state_compute.root_hash(),
+			version,
+		);
+		let block_executor_clone = block_executor.clone();
+		tokio::task::spawn_blocking(move || {
+			block_executor_clone.commit_blocks(vec![block_id], ledger_info_with_sigs)
+		}).await??;
+
+		// commit mempool transactions in batches of size 16
+		for chunk in mempool_transactions.chunks(16) {
+			let mut core_mempool = self.core_mempool.write().await;
+			for transaction in chunk {
 				match transaction.clone().into_inner() {
```

Comment:
We also want to trace here when the transaction is confirmed.

Reply:
Do we need a span around the `commit_transaction` call, or is it expected to be quick?